carbon-design-system/carbon-addons-iot-react
# fix(hotspoteditordatasourcetab): fixed custom labels for multiselect

Closes #2391

**Summary**
Fixes issue where datasource items with custom labels did not show up as selected in the multiselect dropdown of the HotspotEditorDatasourceTab. Also made sure that datasource item label changes are reflected in the dropdown even if they were not present when component loaded.

**Change List (commits, features, bugs, etc)**
* Changed data source items from being rendered with ID to be rendered with Label 
* The initialSelectedItems must completely match the available items for the preselection to work, so the available items have been modified to copy any label changes from the initialSelectedItems.
* Removed unnecessary use of `itemToString`
* Modified the dropdown key generation so that changes to the item labels are picked up. 
* Added a custom label to the HotspotEditorDatasourceTab story
* Added and improved unit tests

**Acceptance Test (how to verify the PR)**

1. Go to https://deploy-preview-2428--carbon-addons-iot-react.netlify.app?path=/story/2-watson-iot-experimental-☢%EF%B8%8F-hotspoteditormodal-hotspoteditordatasourcetab--with-preset-values
2. Click the dropdown
3. Verify that the option "Temperature in Celsius" (which has a custom label) is selected
4. Go to https://deploy-preview-2428--carbon-addons-iot-react.netlify.app?path=/story/2-watson-iot-experimental-☢%EF%B8%8F-hotspoteditormodal--width-existing-hotspots
5. Click on the "My device" hot spot
6. Click on the Data source tab
7. Click the data items dropdown, verify that the option "Custom Label temp" is selected
8. Edit the label and verify that the change is reflected in the data items Dropdown

**Regression Test (how to make sure this PR doesn't break old functionality)**
Test all HotspotEditorModal stories
